proposed/http-message.md

```diff
@@ -97,6 +100,14 @@ echo $message->getHeader('foo');
 // Outputs: baz
 ```
 
+Despite that headers may be retrieved case-insenstively, the original case
+MUST be preserved by the implementation, in particular when retrieved with
+`getHeaders()`.
+
+Non-conforming HTTP applications may depend on a certain case, so it is useful
+for a user to be able to dictate the case of the HTTP headers when creating a 
+request or response.
```

Comment:
I'd argue that this information should also be added to the interface definition itself -- under `withHeader()`, `withAddedHeader()`, and `getHeaders()`.